### PR TITLE
Use InMemory data storage on top of SqLite

### PR DESCRIPTION
**Pull request by hishamco**
Fixes issue #56 

/cc @sebastienros

### CODE REVIEW

**Comment by sebastienros** [comment]
Thank you. Can you also add a test class like the Sqlite one?
Also, the change doesn't compile (see CI)

**Comment by sebastienros** [comment]
When you add the test class, also update the appveyor.yml file to opt in for this test. Look at the end of the file, this will be obvious.

**Comment by hishamco** [comment]
There was a tiny issue before, I fix it and added the test too, nothing but the CI shows an error after adding the test

Any Idea?

**Comment by sebastienros** [comment]
I will test this locally, we might need to set the ReuseConnection argument to true, or the data is lost after each session.

**Comment by hishamco** [comment]
Strange!! It was working with the same `ConnestionString` before, have a look to the old commit https://github.com/sebastienros/yessql/commit/1d643b1f89b60592e1e6e2852bd31dc0b46e688d

**Comment by sebastienros** [comment]
The _new_ unit test class is failing.

**Comment by hishamco** [comment]
I'm getting **no such table: Identifiers** any idea?!!

**Comment by sebastienros** [comment]
I just pushed the fixes that you need. I tried to create it myself and all tests pass.

You just have to rebase on master, then add your extension method and don't forget to pass `true` for `sharedConnection`. Then it will work.

**Comment by hishamco** [comment]
Well done!! thanks @sebastienros

File: test/YesSql.Tests/InMemoryTests.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            catch { }
+
+            try
+            {
+                builder.DropTable("Collection1_Content");
+            }
+            catch { }
+        }
+
+        [Fact(Skip = "ReadCommitted is not supported by Sqlite")]
```
Comment (sebastienros) [inline test/YesSql.Tests/InMemoryTests.cs]: Have you tried? Maybe it works in memory.

File: test/YesSql.Tests/InMemoryTests.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        public override Task ShouldReadCommittedRecords()
+        {
+            return base.ShouldReadCommittedRecords();
+        }
+
+        [Fact(Skip = "Sqlite doesn't support concurrent writers")]
```
Comment (sebastienros) [inline test/YesSql.Tests/InMemoryTests.cs]: Same comment here

File: test/YesSql.Tests/InMemoryTests.cs (hunk 1 above)
Comment (hishamco) [inline test/YesSql.Tests/InMemoryTests.cs]: The test is failed 😞

**Comment by hishamco** [comment]
I update the PR to make `sharedConnection` as optional parameter

**Pull request merged**